op-chain-ops/script/script.go

```diff
@@ -400,6 +417,18 @@ func (h *Host) unwindCallstack(depth int) {
 						h.log.Trace("Broadcast is active, ignoring static-call.")
 					} else {
 						currentCallFrame := h.callStack[len(h.callStack)-1]
+
+						// Have to handle vm.CALLs here since the VM does not
+						// increase nonces for calls. Broadcasting changes this
+						// behavior, so we have to manually increase the nonce.
+						if parentCallFrame.LastOp == vm.CALL {
+							caller := currentCallFrame.Ctx.Caller()
+							h.state.SetNonce(
+								caller,
+								h.state.GetNonce(caller)+1,
+							)
+						}
```

Comment:
The nonce increment should happen when the call happens, not when it wraps up. It's an issue with the `overrideCaller`; when we prank the sender, then the nonce is not incremented on the right account, since geth does the nonce management outside of the `Call`/`Create`/etc. code that does the work and has the override.

Reply:
Left  a TODO in https://github.com/ethereum-optimism/optimism/pull/11839 where the nonce stuff goes